operations/node_management/Swap_a_Compute_Blade_with_a_Different_System_Using_SAT.md

```diff
@@ -0,0 +1,458 @@
+   ncn# sat swap blade -a enable --src-mapping ethernet-interface-mappings-src.json --dst-mapping ethernet-interface-mappings-x1005c0s3-2022-01-01.json x10005c0s3
+   ```
+
+### Power on and boot the nodes
+
+1. Power on and boot the nodes.
+
+   Use `sat bootsys` to power on and boot the nodes. Specify the appropriate BOS template for the node type.
+
+   ```bash
+   ncn# BOS_TEMPLATE=cos-2.0.30-slurm-healthy-compute
+   ncn# sat bootsys boot --stage bos-operations --bos-limit x1005c0s3 --recursive --bos-templates $BOS_TEMPLATE
+   ```
+
+### Check firmware
+
+1. Validate the firmware.
+
+   Verify that the correct firmware versions are present for the node BIOS, node controller (nC), NIC mezzanine card (NMC), GPUs, and so on.
+
+1. If necessary, update the firmware.
+
+   Review the [FAS Admin Procedures](../firmware/FAS_Admin_Procedures.md) and [Update Firmware with FAS](../firmware/Update_Firmware_with_FAS.md) procedure.
+
+   ```bash
+   ncn# cray fas actions create CUSTOM_DEVICE_PARAMETERS.json
+   ```
+
+### Check DVS
+
+There should be a `cray-cps` pod (the broker), three `cray-cps-etcd` pods and their waiter, and at least one `cray-cps-cm-pm` pod. Usually there are two `cray-cps-cm-pm` pods:
+one on `ncn-w002` and one on another worker node.
+
+1. Check the `cray-cps` pods on worker nodes and verify they are `Running`.
+
+   ```bash
+   ncn# kubectl get pods -Ao wide | grep cps
+   ```
+
+   Example output:
+
+   ```text
+   services   cray-cps-75cffc4b94-j9qzf    2/2  Running   0   42h 10.40.0.57  ncn-w001
+   services   cray-cps-cm-pm-g6tjx         5/5  Running   21  41h 10.42.0.77  ncn-w003
+   services   cray-cps-cm-pm-kss5k         5/5  Running   21  41h 10.39.0.80  ncn-w002
+   services   cray-cps-etcd-knt45b8sjf     1/1  Running   0   42h 10.42.0.67  ncn-w003
+   services   cray-cps-etcd-n76pmpbl5h     1/1  Running   0   42h 10.39.0.49  ncn-w002
+   services   cray-cps-etcd-qwdn74rxmp     1/1  Running   0   42h 10.40.0.42  ncn-w001
+   services   cray-cps-wait-for-etcd-jb95m 0/1  Completed
+   ```
+
+1. SSH to each worker node running CPS/DVS and run `dmesg -T`.
+
+   Ensure that there are no recurring `"DVS: merge_one"` error messages shown. These error messages indicate that DVS
+   is detecting an IP address change for one of the client nodes.
+
+   ```bash
+   ncn# dmesg -T | grep "DVS: merge_one"
+   ```
+
+   ```text
+   [Tue Jul 21 13:09:54 2020] DVS: merge_one#351: New node map entry does not match the existing entry
+   [Tue Jul 21 13:09:54 2020] DVS: merge_one#353:   nid: 8 -> 8
+   [Tue Jul 21 13:09:54 2020] DVS: merge_one#355:   name: 'x3000c0s19b1n0' -> 'x3000c0s19b1n0'
+   [Tue Jul 21 13:09:54 2020] DVS: merge_one#357:   address: '10.252.0.26@tcp99' -> '10.252.0.33@tcp99'
+   [Tue Jul 21 13:09:54 2020] DVS: merge_one#358:   Ignoring.
+   ```
+
+1. Make sure the Configuration Framework Service (CFS) finished successfully. Review *HPE Cray Operating System Administration Guide: CSM on HPE Cray EX Systems (S-8024)*.
```

Comment:
> We should replace the publication with a link to one of the CFS docs in the repo. Will need to figure out which one, though. @rbak-hpe do you know if we have anything in docs-csm that covers generally how to take a node xname and see if CFS completed successfully on it? If not, we'll probably want to add that, since it seems like a common task.
> 
> If we don't have anything for this currently in docs-csm then there's no need to hold up this PR for it. We'll handle that with a new ticket.

Ryan is out this week. I'll open a new ticket for this piece. New ticket is: https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-4751